File: Data Structures & Algorithms/n-th-tribonacci-number/submission-3.cpp
class Solution {
public:
    int tribonacci(int n) {
        if (n <= 2) {
            return n == 0 ? 0 : 1;
        }
        int v0=0,v1=1,v2=1;
        for(int i=3;i<=n;i++){
            int temp=v0+v1+v2;
            v0=v1;
            v1=v2;
            v2=temp;
        }
        return v2;
    }
};

File: Data Structures & Algorithms/clone-graph/submission-3.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val = 0, neighbors = None):
        self.val = val
        self.neighbors = neighbors if neighbors is not None else []
"""

class Solution:
    def cloneGraph(self, node: Optional['Node']) -> Optional['Node']:
        visited={}
        def dfs(node):
            if not node:
                return node
            if node.val in visited:
                return visited[node.val]
            temp=Node(node.val)
            visited[node.val]=temp
            for nei in node.neighbors:
                temp.neighbors.append(dfs(nei))
            return temp
        return dfs(node) if node else None